### PR TITLE
Java

### PR DESCRIPTION
**Pull request by olajhidey**
This PR consist of documentation for the compatibility Java Library.
For now documentation on Voice, Messaging still working on the rest though

### CODE REVIEW

File: docs/java/index.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
+---
+sidebar_position: 0
+---
+
+# Compatibility API for Java
+
+## Getting Started
+
+An  Java API wrapper  allows for the rapid development and integration of Java-based applications with the SignalWire API.
```
Comment (danieleds) [inline docs/java/index.mdx]:
```suggestion
The Java API wrapper allows for the rapid development and integration of Java-based applications with the SignalWire API.
```

**Comment by danieleds** [comment]
@olajhidey could you run prettier on the files? Either via the vscode extension, or using `npx prettier --write docs/java`

File: docs/java/reference/messaging/MessageRepository.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+| ---- | ---- | ----------- |
+| `projectId` | `String` | ProjectId gotten from your Signalwire space e.g `a10d8a9f-2166-4e82-56ff-118bc3a4840f` |
+| `apiToken` | `String` | Api token generated from your Signalwire dashboard e.g `PT9e5660c101cd140a1c93a0197640a369cf5f16975a0079c9`|
+| `spaceName` | `String` | Name of your space e.g `testing` |
+
+## Methods
+
+### create
+
+▸**create(**`String to`, `String from`, `String body`**)**:`NewMessageResponse`
```
Comment (danieleds) [inline docs/java/reference/messaging/MessageRepository.mdx]: Can we use the same style used in the `signalwire/docs` repo? E.g.

▸ **create**(String `to`, String `from`, String `body`): NewMessageResponse

    ▸ **create**(String `to`, String `from`, String `body`): NewMessageResponse

(notice the different spaces / bold)

File: docs/java/reference/messaging/MessageRepository.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+| `body` | `String` | The text of the message. Up to 1600 characters long and can be null if no message was sent. |
+
+#### Returns
+`NewMessageResponse`
+
+### create(MessageRequest request)
+
+▸**create(**`MessageRequest request`**)**:`NewMessageResponse`
+
+Create a message with additional information
```
Comment (danieleds) [inline docs/java/reference/messaging/MessageRepository.mdx]: For consistency make sure to always add the period at the end of sentences

File: docs/java/reference/messaging/MessageRepository.mdx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+### getMessages
+
+▸**getMessages()**:`Messages`
+
+Get all the messages on your Signalwire project
+
+#### Returns
+`Messages`
+
+### getMessages(MessageFilter filter)
```
Comment (danieleds) [inline docs/java/reference/messaging/MessageRepository.mdx]: to make the title shorter, could we change this (and any other similar case) to just display the type of the parameter? E.g.

    ### getMessages(MessageFilter)

just for the header (the line below is fine)

File: docs/java/reference/messaging/MessageRepository.mdx [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+**Parameters**
+
+|Name|Type|Description|
+| --- | --- | ---|
+| `to` | `String` | The phone number in E.164 format that received the message |
+| `from` | `String` | The phone number in E.164 format. For inbound messages, this will be the remote phone number who sent the message. |
+| `body` | `String` | The text of the message. Up to 1600 characters long and can be null if no message was sent. |
+
+#### Returns
+`NewMessageResponse`
```
Comment (danieleds) [inline docs/java/reference/messaging/MessageRepository.mdx]: Make sure to link all class / member names to their respective page. This should link to NewMessageResponse.mdx

**Comment by danieleds** [comment]
Changes are great! I added a few suggestions here and there to ensure consistency

**Comment by olajhidey** [comment]
made some addition and fixed some of the suggestions in your previous comment. looking forward to your reply

**Review by danieleds — APPROVED** [review]

**Pull request merged**